### PR TITLE
docs: Update warning about Python 3.9.0 UB, now that 3.9.1 has been released

### PR DESCRIPTION
**Pull request by YannickJadoul**
## Description

As the title says. I've been reasonably conservative, for now, since 3.9.1 is still new, but I would expect that in a few months, we could reduce this further?

### CODE REVIEW

**Review by bstaletic — APPROVED** [review]
Merry ~~christmas~~ python 3.9.1.

File: README.rst [hunk 1]
```diff
@@ -13,16 +13,16 @@
...
+   interpreter shutdown (but could also destroy your data. **You have been
    warned.**)
 
-   We recommend that you wait for Python 3.9.1 slated for release in December,
-   which will include a `fix <https://github.com/python/cpython/pull/22670>`_
-   that resolves this problem.  In the meantime, please update to the latest
-   version of pybind11 (2.6.0 or newer), which includes a temporary workaround
-   specifically when Python 3.9.0 is detected at runtime.
+   We recommend that you update to the latest patch release of Python (3.9.1),
+   which includess a `fix <https://github.com/python/cpython/pull/22670>`_
```
Comment (Skylion007) [inline README.rst:22]: typo: "includes" line 22

**Review by Skylion007 — CHANGES_REQUESTED** [review]
Typo

File: README.rst (hunk 1 above)
Comment (YannickJadoul) [reply README.rst]: Thanks! :-)

**Review by Skylion007 — APPROVED** [review]

**Comment by YannickJadoul** [comment]
Thanks, all! :-)

**Pull request merged**